raiden_contracts/contracts/TokenNetwork.sol

```diff
@@ -191,30 +203,27 @@ contract TokenNetwork is Utils {
     /// @notice Sets the channel participant total deposit value.
     /// Can be called by anyone.
     /// @param channel_identifier The channel identifier - mapping key used for `channels`
-    /// @param participant Channel participant who's deposit is being set.
+    /// @param participant Channel participant whose deposit is being set.
     /// @param total_deposit Idempotent function which sets the total amount of
     /// tokens that the participant will have as a deposit.
     function setDeposit(
         uint256 channel_identifier,
         address participant,
         uint256 total_deposit)
+        isOpen(channel_identifier)
+        isParticipant(channel_identifier, participant)
         public
     {
         uint256 added_deposit;
         Channel storage channel = channels[channel_identifier];
         Participant storage participant_state = channel.participants[participant];
 
-        // Channel must be open and participant must be part of the channel.
-        require(participant_state.initialized);
-
-        // Channel cannot be closed
-        require(closing_requests[channel_identifier].settle_block_number == 0);
-
         require(participant_state.deposit < total_deposit);
 
+        // Calculate the actual amount of tokens that will be transferred
         added_deposit = total_deposit - participant_state.deposit;
 
-        // Change the state
+        // Update the participant's channel deposit
         participant_state.deposit += added_deposit;
```

Comment:
Why not update this directly? `participant_state.deposit = total_deposit` ?

Reply:
I was thinking of it more like a measure of security - increase with the actual amount used in the token transfer. If proper checks are in place, either way should be fine.